app/src/main/java/homer/api/Controller.java

```diff
@@ -0,0 +1,25 @@
+package homer.api;
+
+import java.util.Set;
+
+public interface Controller {
+
+    /**
+     * Connects a new device to the controller.
+     * @param device The device to connect.
+     */
+    void connectDevice(Device<?> device);
```

Comment:
Yes, it could be something like `boolean disconnectDevice(id)` ?
Also we should probably review the connectDevice method, not sure if we are meant to pass a `Device` type directly.

Reply:
@aleemont1 what do you think?

Reply:
Why `boolean`?

Reply:
I was thinking to report whether the operation was successful or not, but it's probably not of much use so we can make it `void` too.

Reply:
How about returning the device id? Even for `connectDevice`

Reply:
Okie, seems kegit, void is ok then